src/commands/distribute/lib/mc-fus-uploader/mc-fus-uploader.ts

```diff
@@ -0,0 +1,764 @@
+import {
+  IUploadData,
+  IUploadStatus,
+  McFusUploadState,
+  IProgress,
+  IUploadStats,
+  McFusMessageLevel,
+  IRequiredSettings,
+  IEventSettings,
+  IInitializeSettings,
+  IOptionalSettings,
+  IOnResumeStartParams,
+  McFusFile,
+  LogProperties,
+} from "./mc-fus-uploader-types";
+import * as fs from "fs";
+import fetch from "node-fetch";
+import { MimeTypes } from "./mc-fus-mime-types";
+
+export class McFile implements McFusFile {
+  readonly name: string;
+
+  public constructor(name: string) {
+    this.name = name;
+  }
+
+  get size(): number {
+    const stats = fs.statSync(this.name);
+    return stats["size"];
+  }
+
+  get fileName(): string {
+    return this.name.replace(/^.*[\\\/]/, "");
+  }
+
+  slice(start: number, end: number): Buffer {
+    const data = Buffer.alloc(end - start);
+    const fd = fs.openSync(this.name, "r");
+    fs.readSync(fd, data, 0, data.length, start);
```

Comment:
We should compare speed with async (regular) reading, if there is no diff, then we can keep it

Reply:
On my laptop each slice takes 4ms in average. For a 114mb binary we have 28 parts.
4ms*28 = 112ms the overall delay for 114mb binary. I think we can leave it as is.

Reply:
We don't have a progress bar for the uploading method yet. We trigger a number of uploads in parallel after the slice call, but each of them depends on a corresponding "slice" call. A pseudocode for this would be:
```
for i in concurrentUploads {
chunk = slice()
send(chunk)
}
```